tests/validation/json-schemas/records/records-write.spec.ts

```diff
@@ -123,6 +123,34 @@ describe('RecordsWrite schema definition', () => {
     }).throws('must NOT have additional properties');
   });
 
+  it('should pass if `protocol` exists and its related properties are all present', () => {
+    const invalidMessage = {
```

Comment:
naming this `invalidMessage` is a bit confusing because the test name and inline comments make it seems like this should be a valid message. Do you mean to test that this message will throw if one of the protocol-related fields is missing?